src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -620,8 +630,20 @@ def _sym_is_field(sym):
             else:
                 sym = symbol_table.lookup_with_tag(str(signature))
                 name_lit = Literal(str(signature), CHARACTER_TYPE)
-            self.add_call(program, read_var,
-                          [name_lit, Reference(sym)])
+
+            # TODO #2898: the test for array can be removed if
+            # `is_allocatable` is supported for non-arrays.
```

Comment:
We could add a default implementation that just returns False but that would have the downside that all the Scalar bindings would then have to have "NonAlloc" added to their names. (Or, we retain the check that it is an array too.)

Reply:
I left the tests the way there are, mainly otherwise this patch would become quite large, since all tests would be affected.